### PR TITLE
drivers: wifi: nxp: add Wi-Fi thread configurations

### PR DESCRIPTION
**Pull request by MaochenWang1**
Support configuring Wi-Fi threads stack size and priority through Kconfig.

### CODE REVIEW

**Review by jukkar — DISMISSED** [review]

File: drivers/wifi/nxp/Kconfig.nxp [hunk 1]
```diff
@@ -316,7 +316,8 @@ menu "Wi-Fi driver Stack configurations"
 
 config NXP_WIFI_MON_TASK_STACK_SIZE
 	int "Mon thread stack size"
-	default 1152
+	depends on NXP_RW610
```
Comment (danieldegrasse) [inline drivers/wifi/nxp/Kconfig.nxp:319]: Do we only need the `mon` task on the RW610?

Comment (MaochenWang1) [reply drivers/wifi/nxp/Kconfig.nxp:319]: Yes, the mon task is only used for RW610 to handle host sleep and wifi recovery in Wi-Fi driver.

**Comment by danieldegrasse** [comment]
After some internal discussion, I think it makes sense for us to hold off on merging this for now. The actual code changes to use these Kconfigs still need to be sent to the NXP HAL upstream, and I think it would be best to send these changes with that update, so the Kconfigs can actually be validated by a build.

**Comment by jukkar** [comment]
> I think it makes sense for us to hold off on merging this for now.

Ok, I added DNM label for this PR, please remove it after this is ready.

**Comment by MaochenWang1** [comment]
> After some internal discussion, I think it makes sense for us to hold off on merging this for now. The actual code changes to use these Kconfigs still need to be sent to the NXP HAL upstream, and I think it would be best to send these changes with that update, so the Kconfigs can actually be validated by a build.

Hi @danieldegrasse the related hal_nxp upstream change, please review:
https://github.com/zephyrproject-rtos/hal_nxp/pull/479

**Review by danieldegrasse — DISMISSED** [review]

**Review by danieldegrasse — DISMISSED** [review]

**Review by jukkar — DISMISSED** [review]

**Review by jukkar — DISMISSED** [review]

**Review by dleach02 — APPROVED** [review]

**Review by jukkar — APPROVED** [review]

**Review by mmahadevan108 — APPROVED** [review]

**Pull request merged**